frontend/hooks/useNeedsFunds.ts

```diff
@@ -1,38 +1,37 @@
 import { formatEther, formatUnits } from 'ethers/lib/utils';
+import { isNil } from 'lodash';
 import { useMemo } from 'react';
 
-import { MiddlewareChain, ServiceTemplate } from '@/client';
 import { STAKING_PROGRAMS } from '@/config/stakingPrograms';
 import { getNativeTokenSymbol } from '@/config/tokens';
-import { getServiceTemplate } from '@/constants/serviceTemplates';
+import { SERVICE_TEMPLATES } from '@/constants/serviceTemplates';
 import { TokenSymbol } from '@/enums/Token';
 import { asEvmChainId } from '@/utils/middlewareHelpers';
 
 import { useBalanceContext, useMasterBalances } from './useBalanceContext';
-import { useService } from './useService';
+import { useServices } from './useServices';
 import { useStore } from './useStore';
 
-export const useNeedsFunds = (serviceConfigId?: string) => {
+export const useNeedsFunds = (stakingProgramId?: string) => {
```

Comment:
decided to pass `stakingProgramId` because it's more convenient;
the `useNeedsFunds` is used in two places:
1) on main where we show violet alert; there I pass `selectedStakingProgramId` (which is `active ?? default` - in most cases would use default)
2) on staking contracts list, where we can easily pass stakingProgramId of each contract